lib/operations/operations_api.rb

```diff
@@ -0,0 +1,144 @@
+require 'hawkular'
+require 'websocket-client-simple'
+require 'json'
+
+class Proc
```

Comment:
I am defining a perform `method()` on any block (top lvl) that can have multiple methods accepting another nested blocks. Currently the nested blocks are `.success` and `.failure` which are the actuall callbacks) Code is quite complex, but the resulting API is imho easy to read:
```ruby
client = OperationsClient.new(...)
client.invoke_generic_operation(foo) do |on|
  on.success do |data|
    # do something with data
  end
  on.failure do |error|
    # handle error
  end
end
```

I've added some docs and check that the blocks respond_to? 'perform', actually, the PR was pretty broken by my manual split and git auto merge. Tests were not passing.

I haven't found a way to make it working under the `Hawkular::Operations` module. If the class definition/overriding is inside the `Hawkular::Operations module`, it would be talking about `Hawkular::Operations::Proc` and not the `Proc`